js/digifinex.js

```diff
@@ -0,0 +1,528 @@
+            'last': this.safeFloat (ticker, 'last'),
+            'previousClose': undefined,
+            'change': undefined,
+            'percentage': undefined,
+            'average': undefined,
+            'baseVolume': undefined,
+            'quoteVolume': undefined,
+            'info': ticker,
+        };
+    }
+
+    parseTrade (trade, market) {
+        let timestamp = parseInt (trade['date']) * 1000;
+        let side = trade['type'].toLowerCase ();
+        let price = this.safeFloat (trade, 'price');
+        let amount = this.safeFloat (trade, 'amount');
+        let cost = price * amount;
+        let fee = undefined;
+        return {
+            'id': undefined,
+            'info': trade,
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'symbol': market['symbol'],
+            'type': undefined,
+            'order': undefined,
+            'side': side,
+            'price': price,
+            'amount': amount,
+            'cost': cost,
+            'fee': fee,
+        };
+    }
+
+    async fetchTrades (symbol, since = undefined, limit = 50, params = {}) {
+        let request = {
+            'symbol': symbol,
+        };
+        let response = await this.privateGetTradeDetail (this.extend (request, params));
+        return this.parseTrades (response['data'], request, since, limit);
+    }
+
+    async createOrder (symbol, type, side, amount, price = undefined, params = {}) {
+        // exchange-specific parameters: post_only
+        // post-only: 1: yes, 0: no. A post_only order will only be created as a maker order. If it cannot be placed into order book immediately, it will be cancelled automatically.
+        await this.loadMarkets ();
+        const order = {
+            'symbol': symbol,
+            'price': this.priceToPrecision (symbol, price),
+            'amount': this.amountToPrecision (symbol, amount),
+            'type': side,
+        };
+        if (type === 'market') {
+            throw new InvalidOrder ('market order not supported');
+        }
+        let response = await this.privatePostTrade (this.extend (order, params));
+        let result = {
+            'info': response,
+            'id': response['order_id'],
+            'symbol': symbol,
+            'side': side,
+        };
+        return result;
+    }
+
+    async cancelOrder (id, symbol = undefined, params = {}) {
+        // await this.loadMarkets ();
+        let response = await this.privatePostCancelOrder ({ 'order_id': id });
+        return response;
+    }
+
+    async CancelOrders (ids, symbol = undefined, params = {}) {
+        // maximum 20 IDs supported
+        let response = await this.privatePostCancelOrder ({ 'order_id': ids.join (',') });
+        return response;
+    }
+
+    parseOrder (order, market = undefined) {
+        let side = order['type'];
+        let status = undefined;
+        if (order.status === 0 || order.status === 1) {
```

Comment:
We need to switch the way these associative keys are accessed – to the single-quoted notation, because: https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#accessing-dictionary-keys

Reply:
Also, see how `parseOrderStatus` is implemented in kraken. Let's do the same here, just copy it from kraken and change the string-numbers a bit.